testsuite/MDAnalysisTests/topology/test_mol2.py

```diff
@@ -68,3 +71,99 @@ def test_bond_orders():
     u = mda.Universe(mol2_molecule)
     orders = [bond.order for bond in u.atoms.bonds]
     assert_equal(orders, ref_orders)
+
+
+def test_elements():
+    u = mda.Universe(mol2_molecule)
+
+    assert_equal(
+        u.atoms.elements[:5],
+        np.array(["N", "S", "N", "N", "O"], dtype="U3")
+    )
+
+
+# Test for #2927
+def test_elements_selection():
+    u = mda.Universe(mol2_molecule)
+    ag = u.select_atoms("element S")
+
+    assert_equal(
+        ag.elements,
+        np.array(["S", "S"], dtype="U3")
+    )
+
+
+mol2_wrong_element = """\
+@<TRIPOS>MOLECULE
+FXA101_1
+49 51 1 0 0
+SMALL
+USER_CHARGES
+
+
+@<TRIPOS>ATOM
+  1 N1       6.8420     9.9900    22.7430 N.am  1 Q101  -0.8960
+  2 S1       8.1400     9.2310    23.3330 X.o2  1 Q101   1.3220
+  3 N2       4.4000     9.1300    20.4710 XX.am  1 Q101  -0.3970
+"""
+
+
+def test_wrong_elements_warnings():
```

Comment:
Might have missed it, but could you add a test that satisfies line 189 on the parser? i.e. all elements have not been assigned so the elements attribute is not created?

Reply:
Well spotted, thanks! There was indeed a bug due to the fact that `validated_elements` is a list and not a `np.ndarray`... I fixed it by pre-allocating `validated_elements` as an array (and added the corresponding test).